kolibri_explore_plugin/assets/src/views/SearchPage.vue

```diff
@@ -164,41 +166,41 @@
         return this.query.trim();
       },
       resultChannels() {
-        const { channels } = this.searchResult;
-        if (!channels || !channels.length) {
-          return null;
-        }
+        return kinds
+          .map(k => _.get(this.searchResult, [k, 'channels'], []))
+          .reduce((accumulator, value) => {
+            return _.uniq([...accumulator, ...value], false, 'id');
+          }, []);
+      },
 
-        return _.chunk(this.searchResult.channels, this.columns);
+      resultChannelsColumns() {
+        return _.chunk(this.resultChannels, this.columns);
       },
       resultCards() {
-        const { results } = this.searchResult;
-        if (!results || !results.length) {
-          return null;
-        }
-
-        // Add tags to nodes
-        const nodes = utils.parseNodes(results);
-
-        // Add thumbnails and links to nodes
-        nodes.forEach(node => {
-          const thumbnailUrl = getContentNodeThumbnail(node);
-          node.thumbnail = thumbnailUrl;
-          const base = `/topics/${node.channel_id}`;
-          if (node.kind === 'topic') {
-            node.nodeUrl = `${base}/t/${node.id}`;
-          } else {
-            node.nodeUrl = `${base}/c/${node.id}`;
-          }
+        let groups = kinds.map(k => {
+          const results = _.get(this.searchResult, [k, 'results'], []);
+          // Add tags to nodes
+          const nodes = utils.parseNodes(results);
+          // Add thumbnails and links to nodes
+          nodes.forEach(node => {
+            const thumbnailUrl = getContentNodeThumbnail(node);
+            node.thumbnail = thumbnailUrl;
+            const base = `/topics/${node.channel_id}`;
+            if (node.kind === 'topic') {
+              node.nodeUrl = `${base}/t/${node.id}`;
+            } else {
+              node.nodeUrl = `${base}/c/${node.id}`;
+            }
+          });
+
+          return [k, nodes];
         });
 
-        // Group by content kind
-        const grouped = _.groupBy(nodes, n => n.kind);
-        // Order by number of results
-        const sortedKeys = _.sortBy(Object.keys(grouped), k => grouped[k].length).reverse();
-        const sortedValues = sortedKeys.map(k => grouped[k]);
-        const zipped = _.zip(sortedKeys, sortedValues);
-        return zipped;
+        // Remove empty groups
+        groups = groups.filter(([, nodes]) => nodes.length > 0);
```

Comment:
In the same way you did for the `resultChannels`, you can chain the filter after the map and avoid using a `let` variable.